src/functions/operatorname.js

```diff
@@ -159,3 +160,5 @@ defineFunction({
     mathmlBuilder,
 });
 
+defineMacro("\\operatorname",
+  "\\@ifstar\\operatornamewithlimits\\operatorname@");
```

Comment:
It had not occurred me that it is possible to call `defineMacro()` from a file other than `macros.js`. Intriguing.

Is it wise, though? When I look for a macro, I always look in `macros.js`. I'm not sure that it would occur to me to grep all the source code for `defineMacro`. And thus I could be misled about whether some macro exists.

Reply:
On the other hand, this approach certainly aligns with the [spirit of the functions folder](https://github.com/KaTeX/KaTeX/pull/766#issuecomment-322648814). Maybe I just need a day to think and let it become familiar.

Reply:
For what it's worth, I did a similar thing in not-yet-merged #2952 (though maybe it feels slightly more comfortable in environments over functions?). In both cases, I feel like it's worth keeping logically clustered content together. 

A smaller example is that `environments/array.js`/`cd.js` redefines the `\cr` macro (though not using `defineMacro`).

In general I feel like `macros.js` is getting rather large and would probably be nicer to split it into logical components. The first chunk about defining macros for the sake of macro definitions makes sense, but the definition of various symbols, `\not`, `\dots`, etc. feel like they could belong elsewhere.  What if we moved those to the `functions` directory?  Or maybe it would be better to rename `functions` to `commands` and make explicit that commands can be defined by functions, macros, or a mixture thereof?